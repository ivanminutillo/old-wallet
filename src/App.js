// @flow
import * as React from 'react'
import {
  View
} from 'react-native'
import Asset from './pages/asset'
import Profile from './pages/profile'
import {Router, Switch, Route} from './Routing'
import {compose, withState, withHandlers} from 'recompose'


// ROUTING
const App = () => (
  <Router>
    <View style={{flex: 1}}>
    <Switch>
      <Route exact path='/' component={Asset} />
      {/* <Route exact path='/profile' component={Profile} /> */}
    </Switch>
    </View>
  </Router>
)

const EnhancedWrapper = compose(
  withState('drawerVisibility', 'handleDrawerVisibility', false),
  withHandlers({
    toggleDrawer: props => () => props.handleDrawerVisibility(!props.drawerVisibility)
  }
)
)(App)

export default EnhancedWrapper