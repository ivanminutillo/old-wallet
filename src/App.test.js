import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import renderer from "react-test-renderer";
import App from "./App";

it("renders without crashing", () => {
  const rendered = renderer.create(<App />).toJSON();
  expect(rendered).toBeTruthy();
});

it("matches a snapshot", () => {
  const component = shallow(<App />);

  expect(component.length).toBe(1);
  expect(toJson(component)).toMatchSnapshot();
});
