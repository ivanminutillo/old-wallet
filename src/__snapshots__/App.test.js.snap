// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <StatusBar
    animated={false}
    hidden={true}
    showHideTransition="fade"
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#222",
        "height": 150,
        "justifyContent": "center",
        "padding": 10,
      }
    }
  >
    <TextElement
      h4={true}
      style={
        Object {
          "color": "#aaa",
        }
      }
    >
      React Native Elements
    </TextElement>
    <TextElement
      h5={true}
      style={
        Object {
          "color": "#aaa",
        }
      }
    >
      Cross-Platform Example
    </TextElement>
  </View>
  <ScrollView>
    <Buttons />
  </ScrollView>
</View>
`;
